include/boost/geometry/views/detail/boundary_view/implementation.hpp

```diff
@@ -0,0 +1,411 @@
+    const_iterator end() const { return base_type::end(); }
+};
+
+
+template <typename Geometry, typename Tag = typename tag<Geometry>::type>
+struct views_container_initializer
+{};
+
+template <typename Polygon>
+struct views_container_initializer<Polygon, polygon_tag>
+{
+    typedef polygon_rings_iterator<Polygon> rings_iterator_type;
+
+    static inline std::size_t num_rings(Polygon const& polygon)
+    {
+        return geometry::num_interior_rings(polygon) + 1;
+    }
+
+    template <typename BoundaryView>
+    static inline void apply(Polygon const& polygon, BoundaryView* views)
+    {
+        std::uninitialized_copy(rings_iterator_type(polygon),
+                                rings_iterator_type(polygon, true),
+                                views);
+    }
+};
+
+template <typename MultiPolygon>
+struct views_container_initializer<MultiPolygon, multi_polygon_tag>
+{
+    typedef typename boost::mpl::if_
+        <
+            boost::is_const<MultiPolygon>,
+            typename boost::range_value<MultiPolygon>::type const,
+            typename boost::range_value<MultiPolygon>::type
+        >::type polygon_type;
+
+    typedef polygon_rings_iterator<polygon_type> rings_iterator_type;
+
+    static inline std::size_t num_rings(MultiPolygon const& multipolygon)
+    {
+        return geometry::num_interior_rings(multipolygon)
+            + static_cast<std::size_t>(boost::size(multipolygon));
+    }
+
+    template <typename BoundaryView>
+    static inline void apply(MultiPolygon const& multipolygon,
+                             BoundaryView* views)
+    {
+        BoundaryView* cur_it = views;
+        for (typename boost::range_iterator<MultiPolygon>::type
+                 it = boost::begin(multipolygon);
+             it != boost::end(multipolygon);
+             ++it)
+        {
+            cur_it = std::uninitialized_copy(rings_iterator_type(*it),
+                                             rings_iterator_type(*it, true),
+                                             cur_it);
+        }
+    }
+};
+
+
+template <typename Areal>
+class areal_boundary
+{
+    typedef boundary_view<typename ring_type<Areal>::type> boundary_view_type;
+    typedef views_container_initializer<Areal> initializer;
+
+    inline void initialize_views(Areal const& areal)
+    {
+        std::pair<boundary_view_type*, std::ptrdiff_t> result
+            = std::get_temporary_buffer<boundary_view_type>(m_num_rings);
+
+        if ( result.second < m_num_rings )
+        {
+            throw std::bad_alloc();
+        }
+
+        m_views = result.first;
+        try
+        {
+            initializer::apply(areal, m_views);
+        }
+        catch (...)
+        {
+            clear();
+            throw;
+        }
+    }
+
+    inline void clear()
+    {
+        for (std::ptrdiff_t i = 0; i < m_num_rings; ++i)
+        {
+            m_views[i].~boundary_view_type();
+        }
+        std::return_temporary_buffer(m_views);
+    }
```

Comment:
At least in STL `clear` name is used for a Container's function removing elements from it. In the case of `vector` it destroys the elements but keeps the allocated memory but in the case of a list it deallocate the memory as well. So AFAIU it's a higher-level abstraction than this "deinitialization" routine. Using `clear` name may be confusing here, at least for me is counter-intuitive. The name should probably also somehow refer to the naming of initialization counterpart. So it could be e.g. `deinitializer::apply()` or `deinitialize()`.